internal/provider/data_source_gitlab_groups_test.go

```diff
@@ -0,0 +1,130 @@
+//go:build acceptance
+// +build acceptance
+
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataSourceGitlabGroups_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+	rInt2 := acctest.RandInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGitlabGroupsConfig(rInt, rInt2),
+				Check: resource.ComposeTestCheckFunc(
```

Comment:
For our provider tests, we perform setup prior to calling `resource.ParallelTest()`, to isolate the resource being tested. Example: https://github.com/gitlabhq/terraform-provider-gitlab/blob/2f06bac4d85ce66bbffdd6b9dc60d319782166be/internal/provider/resource_gitlab_project_approval_rule_test.go#L30

Could you make this change please? This should drastically reduce the number of steps and amount of TF config you need.

Reply:
Thank you for the suggestion. 
I need to create groups with specific names for testing the search of the data_source. 
With groups created by testAccCreateGroups, it is only possible to search for "acctest-group". The result contains also groups from parallel running tests and thus the number of results can not be verified.
To fix this I implemented a helper function testAccCreateGroupsWithPrefix and refactored the test with this function.

I hope that matches your intention.

Reply:
Thanks